`CONFIG_LOG_TIMESTAMP_USE_REALTIME` no longer works

### Describe the bug

Setting `CONFIG_LOG_TIMESTAMP_USE_REALTIME=y` no longer works because

```txt
/<path>/zephyr/subsys/logging/log_core.c: In function 'default_rt_get_timestamp':
/<path>/zephyr/subsys/logging/log_core.c:245:25: error: storage size of 'tspec' isn't known
  245 |         struct timespec tspec;
      |                         ^~~~~
/<path>/zephyr/subsys/logging/log_core.c:245:25: warning: unused variable 'tspec' [-Wunused-variable]
/<path>/zephyr/subsys/logging/log_core.c:250:1: warning: control reaches end of non-void function [-Wreturn-type]
  250 | }
      | ^
```

This was likely introduced in 7868b4d54d123fbaeccea439fa30c4ea78515648

### Regression

- [x] This is a regression.

### Steps to reproduce

1. Any project with `CONFIG_LOG=y`, add `CONFIG_LOG_TIMESTAMP_USE_REALTIME=y`
2. `west build -p --board esp32s3_devkitc/esp32s3/procpu`


### Relevant log output

```shell

```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: macOS 15.5 
- Toolchain: Zephyr SDK 0.17.0
- Zephyr codebase at d41716f63c5c86c811e06559c25a668a9beb1d5a

### Additional Context

Adding `#include <time.h>` to `subsys/logging/log_core.c` fixes it but I am unsure if this should be the case

--- a/subsys/logging/log_core.c
+++ b/subsys/logging/log_core.c
@@ -3,6 +3,7 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+#include <time.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/logging/log_backend.h>
 #include <zephyr/logging/log_ctrl.h>

